test/test_selectors.py

```diff
@@ -0,0 +1,622 @@
+    # raise the exception.
+    @skipUnless(hasattr(signal, "alarm"), "Platform doesn't have signal.alarm()")
+    def test_select_interrupt_exception(self):
+        s = self.make_selector()
+        rd, wr = self.make_socketpair()
+        s.register(rd, selectors.EVENT_READ)
+
+        class AlarmInterrupt(Exception):
+            pass
+
+        def alarm_exception(*args):
+            raise AlarmInterrupt()
+
+        sigalrm_handler = signal.signal(signal.SIGALRM, alarm_exception)
+        self.addCleanup(signal.signal, signal.SIGALRM, sigalrm_handler)
+        self.addCleanup(signal.alarm, 0)
+
+        # Start the timer for the interrupt.
+        signal.alarm(1)
+
+        t = monotonic()
+        self.assertRaises(AlarmInterrupt, s.select, 2.0)
+        self.assertLess(monotonic() - t, 1.2)
+
+    def test_fileno(self):
+        s = self.make_selector()
+        if hasattr(s, "fileno"):
+            fd = s.fileno()
+            self.assertTrue(isinstance(fd, int))
+            self.assertGreaterEqual(fd, 0)
+
+    # According to the psutil docs, open_files() has strange behavior
+    # on Windows including giving back incorrect results so to
+    # stop random failures from occurring we're skipping on Windows.
+    @skipIf(sys.platform == "win32", "psutil.Process.open_files() is unstable on Windows.")
+    def test_leaking_fds(self):
+        proc = psutil.Process()
+        before_fds = len(proc.open_files())
+        s = self.make_selector()
+        s.close()
+        after_fds = len(proc.open_files())
+        self.assertEqual(before_fds, after_fds)
+
+    @skipUnless(sys.platform == "win32", "This is the same as test_leaking_fds but for Windows")
+    def test_leaking_fds_windows(self):
+        proc = psutil.Process()
+        before_fds = len(proc.open_files())
+        s = self.make_selector()
+        s.close()
+        after_fds = len(proc.open_files())
+        if before_fds != after_fds:
+            self.skipTest("Leaked fd but test was on Windows.")
+
+class ScalableSelectorMixin:
+    """ Mixin to test selectors that allow more fds than FD_SETSIZE """
+    @skipUnless(resource, "Could not import the resource module")
+    @skipUnless(sys.platform != "darwin", "Can't run on Mac OS due to RINFINITE hard limit.")
+    def test_above_fd_setsize(self):
+        # A scalable implementation should have no problem with more than
+        # FD_SETSIZE file descriptors. Since we don't know the value, we just
+        # try to set the soft RLIMIT_NOFILE to the hard RLIMIT_NOFILE ceiling.
+        soft, hard = resource.getrlimit(resource.RLIMIT_NOFILE)
+
+        try:  # If we're on a *BSD system, the limit tag is different.
+            _, bsd_hard = resource.getrlimit(resource.RLIMIT_OFILE)
+            if bsd_hard < hard:
+                hard = bsd_hard
+
+        # NOTE: AttributeError resource.RLIMIT_OFILE is not defined on Mac OS.
+        except (OSError, resource.error, AttributeError):
+            pass
```

Comment:
Ah, I forgot to change that to something else. So in the actual selectors they test on some Mac OS systems not all, I need to change that so we do the same. Then there will be a need for this.